networkstatsquerier/fetcher/fetcher.go

```diff
@@ -6,28 +6,36 @@ import (
 	"os"
 )
 
-// Keyword is a struct for each keyword
+// OrgKeyword is a struct for each keyword
 type OrgKeyword struct {
-	Kind  string `json:"Kind"`
-	Value string `json:"Value"`
+	Kind  string
+	Value string
 }
 
 // EndpointEntry is a struct for each entry of data pulled out of the EndpointSources file
 type EndpointEntry struct {
-	OrganizationName     string       `json:"OrganizationName"`
-	FHIRPatientFacingURI string       `json:"FHIRPatientFacingURI"`
-	Type                 string       `json:"Type"`
-	Keywords             []OrgKeyword `json:"Keywords"`
+	OrganizationName     string
+	FHIRPatientFacingURI string
+	ListSource           string
+	Type                 string
+	Keywords             []OrgKeyword
 }
 
 // ListOfEndpoints is a structure for the whole EndpointSources file
 type ListOfEndpoints struct {
-	Entries []EndpointEntry `json:"Entries"`
+	Entries []EndpointEntry
+}
+
+// Endpoints is an interface that every endpoint list can implement to parse their list into
+// the universal format ListOfEndpoints
+type Endpoints interface {
+	GetEndpoints(interface{}) (ListOfEndpoints, error)
 }
 
 // GetListOfEndpoints parses a list of endpoints out of the file at the provided path
 func GetListOfEndpoints(filePath string) (ListOfEndpoints, error) {
 	var result ListOfEndpoints
+	var initialList map[string]interface{}
```

Comment:
In terms of flow - here are my thoughts (completely up for discussion).

We want:
* to be able to easily use this when we start querying from the hosting sites
* to be able to have endpoints from a set of vendors that we know how to process
* endpoints from generic sources (vendors we don't know how to process)

Consider an enum for the known vendors:

https://programming.guide/go/define-enumeration-string.html
https://www.ribice.ba/golang-enums/

(pseudo code)
```
enum Sources {
  Cerner: "cerner url",
  Epic: "epic url"
  ...
}
```

And two `GetListOfEndpoints` methods:

```
// sends the rawendpts to the associated parser and assigns the associated url as the source
func GetListOfEndpointsKnownSource(rawendpts []byte, source Source)

// send sthe rawendpts to the default parser and assigns the given source string as the source
func GetListOfEndpoints(rawendpts []byte, source string)
```

Use an enum so we don't have to have people fat finger the urls for known sources/remember them/etc.


I'm open to other suggestions!

Taking in a byte array rather than a file path makes this easily usable for when we query from an endpoint.
We could add a convenience function that still permits you to send in the file path, which will then extract the byte array and then call one of these functions.

Reply:
I like this, it seems like it'll make implementing pulling the endpoint lists from sites a lot easier in the future! i'll try implementing it and let you know if i come across any problems